blockchain-source/src/cardano/event.rs

```diff
@@ -157,3 +145,51 @@ impl<Tip> GetNextFrom for CardanoNetworkEvent<BlockEvent, Tip> {
         }
     }
 }
+
+impl BlockEvent {
+    pub(crate) fn from_serialized_block(raw_block: SerializedBlock) -> anyhow::Result<Self> {
+        let block = raw_block
+            .unserialize()
```

Comment:
btw is there any chance to deserialize just the header? I've skimmed the library, seems like no way, but better to ask than not :)

Reply:
Well... Not directly I think. We could probably do it ourselves by using the cbored types directly. I think we probably want to verify the structure at least once though. 

And it's not like the whole block is actually parsed, the whole structure is traversed, but nested things like transactions are deserialized lazily, and just kept as arrays until we need it, so it's not a full deserialization.